packages/@expo/config-types/src/ExpoConfig.ts

```diff
@@ -239,6 +239,10 @@ export interface ExpoConfig {
    * Specifies the JavaScript engine for apps. Supported only on EAS Build. Defaults to `hermes`. Valid values: `hermes`, `jsc`.
    */
   jsEngine?: 'hermes' | 'jsc';
+  /**
+   * Enable React Native new architecture.
+   */
+  newArchEnabled?: boolean;
```

Comment:
please remember to create a pr in universe to update the unversioned schema and then generate ExpoConfig.ts here